app/src/main/java/org/oppia/app/utility/ClickableAreasImage.kt

```diff
@@ -0,0 +1,112 @@
+import com.github.chrisbanes.photoview.PhotoViewAttacher
+import org.oppia.app.R
+import org.oppia.app.player.state.ImageRegionSelectionInteractionView
+import kotlin.math.roundToInt
+
+class ClickableAreasImage(
+  private val imageView: ImageRegionSelectionInteractionView,
+  private val parentView: FrameLayout,
+  private val listener: OnClickableAreaClickedListener
+) : OnPhotoTapListener {
+  private val attacher: PhotoViewAttacher = PhotoViewAttacher(imageView)
+
+  init {
+    attacher.setOnPhotoTapListener(this)
+  }
+
+  /**
+   * Called when an image is clicked.
+   *
+   * @param view the original view on which the tap/click occurs.
+   * @param x the relative x coordinate according to image
+   * @param y the relative y coordinate according to image
+   */
+  override fun onPhotoTap(view: ImageView, x: Float, y: Float) {
+    val clickableAreaIndex = getClickableAreaOrDefault(x, y)
+    parentView.forEachIndexed { index: Int, tappedView: View ->
+      if (index > 0) {
+        tappedView.isVisible = index == clickableAreaIndex + 1
+        listener.onClickableAreaTouched(
+          imageView.getClickableAreas()[clickableAreaIndex].label
+        )
+      }
+    }
+  }
+
+  private fun getClickableAreaOrDefault(x: Float, y: Float): Int {
+    return imageView.getClickableAreas().indexOfFirst {
+      isBetween(it.region.area.upperLeft.x, it.region.area.lowerRight.x, x) &&
+        isBetween(it.region.area.upperLeft.y, it.region.area.lowerRight.y, y)
+    }
+  }
+
+  /** Return whether a point lies between two points.*/
+  private fun isBetween(start: Float, end: Float, actual: Float): Boolean {
+    return actual in start..end
+  }
+
+  /** Get X co-ordinate scaled according to image.*/
+  private fun getXCoordinate(x: Float): Float {
+    val rect = attacher.displayRect
+    return (x * rect.width()) + rect.left
+  }
+
+  /** Get Y co-ordinate scaled according to image.*/
+  private fun getYCoordinate(y: Float): Float {
+    val rect = attacher.displayRect
+    return (y * rect.height()) + rect.top
+  }
+
+  fun addViews(isAccessible: Boolean) {
+    parentView.let {
+      imageView.getClickableAreas().forEachIndexed { labelIndex, clickableArea ->
+        val imageRect = RectF(
+          getXCoordinate(clickableArea.region.area.upperLeft.x),
+          getYCoordinate(clickableArea.region.area.upperLeft.y),
+          getXCoordinate(clickableArea.region.area.lowerRight.x),
+          getYCoordinate(clickableArea.region.area.lowerRight.y)
+        )
+        val layoutParams = FrameLayout.LayoutParams(
+          imageRect.width().roundToInt(),
+          imageRect.height().roundToInt()
+        )
+        val newView = View(it.context)
+
+        newView.x = imageRect.left
+        newView.y = imageRect.top
+        newView.isClickable = true
+        newView.isFocusableInTouchMode = true
+        newView.isFocusable = true
+        newView.layoutParams = layoutParams
+        newView.contentDescription = clickableArea.label
```

Comment:
Shouldn't this be the creator-specific content description, not the answer?

Reply:
yes it would be change to a new parameter once it is finalised.

Reply:
We should change this before this PR is submitted--we shouldn't have a content description until we have a real one to specify.

Reply:
for this should i add another field in the proto and use that?

Reply:
done